07-errors.md

```diff
@@ -78,15 +80,17 @@ Standard non-zero exit codes are codes returned by actors to indicate common fai
 | 21    | `ErrSerialization`     | actor failed to serialize or deserialize some state |
 | 22    | `ErrUnhandledMessage`  | actor cannot handle this message                    |
 | 23    | `ErrUnspecified`       | the actor failed with an unspecified error          |
-| 24-31 | reserved               |                                                     |
+| 24    | `ErrAssertionFailed`   | the actor failed some internal assertion            |
+| 25-31 | reserved               |                                                     |
 
 Changes from pre-FVM Filecoin:
 
 1. Added `ExitCode::ErrUnhnadledMessage` (replacing `ExitCode::SysErrInvalidMethod`).
-2. Added `ExitCode::ErrUnspecified`. This generally means "there's a bug and I have no idea what
-   went wrong". The alternative would be for the actor to _panic_ (leading to an
-   `ExitCode::ErrIllegalInstruction` syscall error), but it may be useful to distinguish between
-   "crashing" and "something unknown went wrong".
+2. Added `ExitCode::ErrUnspecified`. This indicates that something unspecified went wrong, but we
+   don't know what the cause is. This is a sane default exit code when accurately determining the
+   underlying error is expensive, unreliable, complex, etc.
+3. Added `ExitCode::ErrAssertionFailed`. This means that the actor violated some internal runtime
+   invariant. _Clean_ panics should exit with this code.
```

Comment:
What is `runtime` here? I don't think this is correct. There are either system-originated exit codes, or user-originated exit codes. This is the latter. Yeah, the builtin-actors `Runtime` emits this, but the concept of a runtime is not standard and it's an implementation choice, so it shouldn't leak here.

Reply:
I said "runtime" to distinguish between some _state_ invariant. But I agree that it's really confusing.